packages/select/src/components/select/suggest.tsx

```diff
@@ -89,26 +89,22 @@ export class Suggest<T> extends React.PureComponent<ISuggestProps<T>, ISuggestSt
         return Suggest as new (props: ISuggestProps<T>) => Suggest<T>;
     }
 
+    public state: ISuggestState<T> = {
+        isOpen: (this.props.popoverProps != null && this.props.popoverProps.isOpen) || false,
```

Comment:
huh, I thought `this.props` would be undefined here, but I guess something changed in the way tsc transpiles ES classes at some point

Reply:
in my experience this has been possible for a very long time. `tsc` generates the constructor around this code for you.

Reply:
Why is this better than writing out the constructor? In my experience explicitly setting the initial state in the constructor is easier to read

Reply:
i don't like the overhead of getting the constructor syntax right in the first place. this is harder to mess up.